Ports gang mode

JTGSZ opened the pull request:
Ports gang from hippiestation.
Fixes a few things in it, changes a few things in port like hud icons.
Parity with 2017 itemlist from /tg/ gang mode.
Fixes regular spraycan decals being below walls thanks to bhijn telling me what was wrong.

:cl: jtgsz
add: ported gang mode
/:cl:

People's nostalgia wanted this back, it might be fun? Its not like it has to join the actual mode cycle. 
I just dumped it all in one folder since that would make it easy to remove/find everything related to it. Hosted and tested.

JTGSZ committed:
ports gang

JTGSZ committed:
woops

left that at 1 instead of 15

deathride58 commented on modular_citadel/code/game/gamemodes/gangs/dominator.dm:
```diff
@@ -0,0 +1,241 @@
+
+/obj/machinery/dominator/Initialize()
+	. = ..()
+	set_light(2)
```
it'd be neat if you could make this inherit the team color on top of reducing the light power to like 0.75 or so

JTGSZ replied:
Light power reduced to 0.75, Light range 2 kept, the mind dominator has a neutral white light tone along with the monitor display since it has no team tied to it yet at this stage though. It basically shows that its awaiting input upon the console overlay.

deathride58 commented on modular_citadel/code/game/gamemodes/gangs/dominator.dm:
```diff
@@ -0,0 +1,241 @@
+		countdown.start()
+		countdown.color = gang.color
+
+		set_light(3)
```
Ditto but with like 0.9 as the light power

JTGSZ replied:
Light power reduced to 0.9, Light range 3 kept, Team color now applies in the lighting along with the new monitor overlay at this stage, showing that its active.

deathride58 commented on modular_citadel/code/game/gamemodes/gangs/dominator.dm:
```diff
@@ -0,0 +1,241 @@
+				vagos.message_gangtools("Enemy takeover attempt detected in [locname]: Estimated [time] minutes until our defeat.",1,1)
+
+/obj/machinery/dominator/proc/excessive_walls_check() // why the fuck was this even a global proc...
+	var/open = FALSE
```
open = 0, open isn't being used as a boolean here

JTGSZ replied:
Open = 0, changed the wall check to a wall check as I noticed it was prone to fucking dominator placement to be a bit more forgiving, but not enough to be worthless.

deathride58 commented on modular_citadel/code/game/gamemodes/gangs/gang_crayon.dm:
```diff
@@ -0,0 +1,236 @@
+#define RANDOM_GRAFFITI "Random Graffiti"
```
Modularization isn't necessary. You can demodularize this safely without issue.

JTGSZ replied:
demodularized and deleted gang_crayon.dm

JTGSZ replied:
unresolving this, I need to redo parts of this.

JTGSZ replied:
Demodularized it, now regular people can use gang spraycans without causing a runtime, and they just fire off regularly. Gang members can use each others spraycans, and regular spraycan decals now can go over walls since you told me about the planes thing.

deathride58 reviewed (commented):
Nitpicks. You can safely demodularize all of the overridden code in this PR, as modularization is no longer necessary for main.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,404 @@
+	id = "pistol_ammo"
+	cost = 10
+	item_path = /obj/item/ammo_box/magazine/m10mm
+
```
Weres the 60 cost 40mm gun with its 20 cost shells?

JTGSZ replied:
Someone would probably stab me If a anti-aircraft gun was in this.

JTGSZ committed:
More changes

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_datums.dm:
```diff
@@ -0,0 +1,139 @@
+	name = "Tunnel"
+	color = "#FF00FF"
+	inner_outfits = list(/obj/item/clothing/under/villain)
+	outer_outfits = list(/obj/item/clothing/suit/poncho/ponchoshame)
```
Isnt ponchoshame no drop?

JTGSZ replied:
Its cursed, yep. Got anything to replace it with? I'll put something else in there, if theres no fun ideas for the tunnel gang suit.

JTGSZ replied:
Swapped the Poncho of shame out with a leather jacket since tunnel gang, changed some clothing around.

JTGSZ committed:
Here we go.

JTGSZ committed:
Solves a few things.

JTGSZ committed:
blunt trauma deconv

JTGSZ committed:
Over-reduced the item costs a bit too much.

Lets do it in tiny increments until It plays somewhat unpredictably for all sides.

JTGSZ committed:
Brute deconv correct

woops

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+/datum/gang_item/weapon/ammo/surplus_ammo
+	name = "Surplus Rifle Ammo"
+	id = "surplus_ammo"
+	cost = 5
```
Prob can be lowerd to 2 or 3 safely

JTGSZ replied:
Lowered seeing as the surplus rifle has seen little use.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+/datum/gang_item/equipment/emp
+	name = "EMP Grenade"
+	id = "EMP"
+	cost = 7
```
Prob should be lowed to 5

JTGSZ replied:
EMP grenades are actually really strong, So it stays for the moment, I feel.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+/datum/gang_item/equipment/implant_breaker
+	name = "Implant Breaker"
+	id = "implant_breaker"
+	cost = 10
```
12 or 15 do to how easy it is to gang a sec officer or head of staff

JTGSZ replied:
Gangs have no good way to stun/capture someone to actually use this, Its seen no prominent use so far. Even then its a option to keep someone alive and playing instead of dead, along with promoting more social interaction.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+/datum/gang_item/equipment/wetwork_boots
+	name = "Wetwork boots"
+	id = "wetwork"
+	cost = 10
```
safe to lower this to 7 or 8 do to not even being armord

JTGSZ replied:
Lowered, seeing as its purpose isn't quite so strong these days.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+	id = "buckshot"
+	cost = 5
+	item_path = /obj/item/storage/box/lethalshot
+
```
Would be a good idea to add in other types of ammo. maybe 5 costs for basic ammo and 20 for tech slug types like dragon or meteor

JTGSZ replied:
Unsure about this one.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+		var/outfit = pick(gang.outer_outfits)
+		if(outfit)
+			var/obj/item/O = new outfit(user.loc)
+			O.armor = O.armor.setRating(melee = 20, bullet = 35, laser = 10, energy = 10, bomb = 30, bio = 0, rad = 0, fire = 30, acid = 30)
```
armor teak 
25 melee 20 laser would help gangs survive sec raids better

JTGSZ replied:
I don't think they should resist lasers a ton seeing as they are ment to fight each other, but I did add 5 to melee and only 5 to laser resists. We will see if this makes people more likely to put on the outer uniform.

Trilbyspaceclone commented on modular_citadel/code/game/gamemodes/gangs/gang_items.dm:
```diff
@@ -0,0 +1,410 @@
+	icon_state = "gangbelt"
+	item_state = "gang"
+	desc = "The belt buckle simply reads 'BAMF'."
+
```
Adding sun-glasses / headphones to deal with flash bangs for 5(Sunglasses) and 3(Headphones) would help against sec raids

JTGSZ replied:
Sunglasses were added to the items list seeing as it just kind of fits with all the other accessories, but I don't think they should deal with flashbangs completely. We will see?

Trilbyspaceclone reviewed (commented):
Few tweaks to give gangs better odds vs sec raids or hold out against other gangs

JTGSZ committed:
Some changes.
